src/dbnode/persist/fs/retriever.go

```diff
@@ -582,6 +582,16 @@ func (req *retrieveRequest) Clone() (xio.SegmentReader, error) {
 	return req.reader.Clone()
 }
 
+func (req *retrieveRequest) DeepClone(
```

Comment:
Hm, what's deep clone used for?

Reply:
I think its so that the Retriever code path doesn't have to clone the bytes again (which have already just been cloned from disk). I need to look through the retriever code path before we merge this, but I do think we want to avoid cloning twice